src/main/java/org/kohsuke/github/GitHubRateLimitHandler.java

```diff
@@ -71,15 +79,29 @@ public void onError(GitHubConnectorResponse connectorResponse) throws IOExceptio
                 throw (InterruptedIOException) new InterruptedIOException().initCause(ex);
             }
         }
+    };
 
-        private long parseWaitTime(GitHubConnectorResponse connectorResponse) {
-            String v = connectorResponse.header("X-RateLimit-Reset");
-            if (v == null)
-                return 60 * 1000; // can't tell, return 1 min
+    /*
+     * Exposed for testability. Given an http response, find the rate limit reset header field and parse it. If no
+     * header is found, wait for a reasonably amount of time.
+     */
+    long parseWaitTime(GitHubConnectorResponse connectorResponse) {
+        String v = connectorResponse.header("X-RateLimit-Reset");
+        if (v == null)
+            return Duration.ofMinutes(1).toMillis(); // can't tell, return 1 min
 
-            return Math.max(1000, Long.parseLong(v) * 1000 - System.currentTimeMillis());
+        // Don't use ZonedDateTime.now(), because the local and remote server times may not be in sync
+        // Instead, we can take advantage of the Date field in the response to see what time the remote server
+        // thinks it is
+        String dateField = connectorResponse.header("Date");
+        ZonedDateTime now;
+        if (dateField != null) {
+            now = ZonedDateTime.parse(dateField, DateTimeFormatter.RFC_1123_DATE_TIME);
+        } else {
+            now = ZonedDateTime.now();
         }
-    };
+        return Math.max(MINIMUM_RATE_LIMIT_RETRY_MILLIS, (Long.parseLong(v) - now.toInstant().getEpochSecond()) * 1000);
```

Comment:
This is now mostly the same as the abuse limit wait parse code you submitted.  Would it make sense to extract this into a shared method?  Perhaps even make it public so others could use it?  Opened #1973 to track this.